sql/src/main/java/io/crate/execution/engine/window/WindowBatchIterator.java

```diff
@@ -190,6 +192,7 @@ public boolean moveNext() {
 
             if (arePeers(currentRowCells, sourceRowCells)) {
                 windowForCurrentRow.add(sourceRowCells);
+                newRowsInCurrentFrame.add(sourceRowCells);
```

Comment:
Would it be possible to just keep the `startIdx` of new rows and then work with views on top of the existing list instead of having to maintain 2 collections?

Reply:
yes, totes